### PR TITLE
Mender growfs data

### PR DESCRIPTION
**Pull request by mirzak**
Follow-up on https://github.com/mendersoftware/meta-mender/pull/822

### CODE REVIEW

File: 03.Devices/02.Yocto-project/01.Partition-configuration/docs.md [hunk 1]
```diff
@@ -104,6 +104,8 @@ When [building a Mender Yocto Project image](../../../artifacts/yocto-project/bu
 | &lt;BOOT&gt; | Store the bootloader.                                   | 16 MB        | `MENDER_BOOT_PART_SIZE_MB`     |
 | `/data`      | Store persistent data, preserved during Mender updates. | 128 MB       | `MENDER_DATA_PART_SIZE_MB`     |
 
+!!! Even though the default size of `MENDER_DATA_PART_SIZE_MB` is `128 MB`, it will try to resized the partition and filesystem image on first boot to the full size of the underlying block device which is also resized to occupy reminder of avaiable blocks on the storage medium. This functionality relies on [systemd-growfs](https://www.freedesktop.org/software/systemd/man/systemd-makefs@.service.html) and take note that only certain filesystem types are supported. See [mender-growfs-data feature](../../../04.Artifacts/01.Yocto-project/02.Image-configuration/01.Features/docs.md) for more information.
```
Comment (kacf) [inline 03.Devices/02.Yocto-project/01.Partition-configuration/docs.md]: * "...try to resize~d~ the partition..."
* "...occupy rem**a**inder of avai**l**able blocks..."

Comment (mirzak) [reply 03.Devices/02.Yocto-project/01.Partition-configuration/docs.md]: Thanks, fixed.

**Comment by estenberg** [comment]
Nice! Could this apply to other OSes as well (e.g. Debian)?

**Comment by drewmoseley** [comment]
In principle I think this could apply to other OSes but each one would likely need to be tested to make sure they don't have something that needs tweaking.

**Comment by mirzak** [comment]
> Nice! Could this apply to other OSes as well (e.g. Debian)?

What @drewmoseley  said, but also I have already "half" implemented this in mender-convert v2,

https://github.com/mirzak/mender-convert/blob/v2/configs/mender_convert_config#L33-L38

But it has a dependency on systemd and systemd version + currently the implementation in mender-convert will not resize the partition and only will resize the filesystem image.

**Pull request merged**